android/app/src/main/java/app/organicmaps/location/TrackRecorder.java

```diff
@@ -0,0 +1,51 @@
+package app.organicmaps.location;
+
+import app.organicmaps.util.Listeners;
+
+public class TrackRecorder
+{
+  private final Listeners<TrackRecordingListener> mListeners = new Listeners<>();
+
+  private static final TrackRecorder INSTANCE = new TrackRecorder();
+
+  private TrackRecorder(){}
+  public static TrackRecorder getInstance()
+  {
+    return INSTANCE;
+  }
+  public interface TrackRecordingListener
+  {
+    void onTrackRecordingStarted();
+    void onTrackRecordingStopped();
+  }
+  public void startTrackRecording()
+  {
+    if(!nativeIsEnabled()) nativeSetEnabled(true);
+    for (TrackRecordingListener listener : mListeners)
+    {
+      listener.onTrackRecordingStarted();
+    }
+    mListeners.finishIterate();
+  }
+  public void stopTrackRecording()
+  {
+    if(nativeIsEnabled()) nativeSetEnabled(false);
+    for (TrackRecordingListener listener : mListeners)
+    {
+      listener.onTrackRecordingStopped();
+    }
+    mListeners.finishIterate();
+  }
+  public void addListener(TrackRecordingListener listener)
+  {
+    mListeners.register(listener);
+  }
+  public void removeListener(TrackRecordingListener listener)
+  {
+    mListeners.unregister(listener);
+  }
+  public static native void nativeSetEnabled(boolean enable);
```

Comment:
Actually after your suggestion to start the foreground service from mwmActivity, i created a listener class which notifies the listener in mwmActivity that track recording has started and then it starts the track recording service.

Reply:
> I don't really buy this idea with listeners in this class. Can we just check the status of Recent Track in MwmActivity.onResume() and start or stop the service? It will be 2 lines of code instead of 100.

There is one UX issue-
Suppose a users enables the recent track recorder from settings and directly exits the app without going to map screen. In that case service will never be started.

If we will place this in onResume() method, service will only turn on when mwmActivity will be resumed  (i.e user will return to map screen) and same for turning it off as well.

Reply:
Since settings has been removed so all these are also gone : )